libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Actions/ForEachElement.cs

```diff
@@ -279,6 +280,18 @@ private JArray ConvertToList(object instance)
                 result = Object2List(jobject);
             }
 
+            if (beginDialog)
```

Comment:
Just checking: have you tried, or is there a test to verify that this does not crash on empty collections on begin dialog?

Reply:
i.e. if you invoke foreach on an empty array

Reply:
added an empty result test